core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -65,4 +80,112 @@ public byte exitStatus() {
+
+        private Builder() {
+        }
+
+        public Builder withArg(final String arg) {
+            this.runtimeOptions = new RuntimeOptions(arg);
+            return this;
+        }
+
+        public Builder withArgs(final String... args) {
+            return withArgs(Arrays.asList(args));
+        }
+
+        public Builder withArgs(final List<String> args) {
+            this.runtimeOptions = new RuntimeOptions(args);
+            return this;
+        }
+
+        public Builder withRuntimeOptions(final RuntimeOptions runtimeOptions) {
+            this.runtimeOptions = runtimeOptions;
+            return this;
+        }
+
+        public Builder withClassLoader(final ClassLoader classLoader) {
+            this.classLoader = classLoader;
+            return this;
+        }
+
+        public Builder withResourceLoader(final ResourceLoader resourceLoader) {
+            this.resourceLoader = resourceLoader;
+            return this;
+        }
+
+        public Builder withClassFinder(final ClassFinder classFinder) {
+            this.classFinder = classFinder;
+            return this;
+        }
+
+        public Builder withBackendSupplier(final BackendSupplier backendSupplier) {
+            this.backendSupplier = backendSupplier;
+            return this;
+        }
+
+        public Builder withGlueSupplier(final GlueSupplier glueSupplier) {
+            this.glueSupplier = glueSupplier;
+            return this;
+        }
+
+        public Builder withFeatureSupplier(final FeatureSupplier featureSupplier) {
+            this.featureSupplier = featureSupplier;
+            return this;
+        }
+
+        public Builder withAdditionalPlugins(final Plugin... plugins) {
+            this.additionalPlugins = Arrays.asList(plugins);
+            return this;
+        }
+
+        public Builder withEventBus(final EventBus eventBus) {
+            this.eventBus = eventBus;
+            return this;
+        }
+
+        public Runtime build() {
+            ResourceLoader resourceLoader = this.resourceLoader != null
+                ? this.resourceLoader
+                : new MultiLoader(this.classLoader);
+
+            ClassFinder classFinder = this.classFinder != null
+                ? this.classFinder
+                : new ResourceLoaderClassFinder(resourceLoader, this.classLoader);
+
+            BackendSupplier backendSupplier = this.backendSupplier != null
+                ? this.backendSupplier
+                : new BackendModuleBackendSupplier(resourceLoader, classFinder, this.runtimeOptions);
+
+            final Plugins plugins = new Plugins(this.classLoader, new PluginFactory(), this.eventBus, this.runtimeOptions);
+            for (final Plugin plugin : additionalPlugins) {
+                plugins.addPlugin(plugin);
+            }
+
+            final RunnerSupplier runnerSupplier = new ThreadLocalRunnerSupplier(this.runtimeOptions, this.eventBus, backendSupplier, this.glueSupplier);
```

Comment:
This should however be customizable.

Reply:
OK, I can just add another `withRunnerSupplier` - but will wait for all feedback first

Reply:
Actually. This was a bit problematic. I've reverted it.

Reply:
Runtime options couldn't escape the builder.